x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -631,6 +647,74 @@ export class RulesClient {
     });
   }
 
+  public async getExecutionLogForRule({
+    id,
+    dateStart,
+    dateEnd,
+    filter,
+    page,
+    perPage,
+    sort,
+  }: GetExecutionLogByIdParams): Promise<IExecutionLogResult> {
+    this.logger.debug(`getExecutionLogForRule(): getting execution log for rule ${id}`);
+    const rule = (await this.get({ id, includeLegacyId: true })) as SanitizedRuleWithLegacyId;
+
+    try {
+      // Make sure user has access to this rule
```

Comment:
**question:** I may not have the full context but I see the `this.authorization.ensureAuthorized` API is also called within the `get(...)` function that is called above (line 660). Do we also need to do a second authorization check here but for `operation: ReadOperations .GetExecutionLog`?

Reply:
Are you asking if this second auth check is required? This second auth check is checking auth for `ReadOperations.GetExecutionLog` (line 667).

Whether it is required...I don't know? We had a special `ReadOperations.GetAlertSummary` that I considered reusing and it might be fine to consolidate them?

Reply:
>Whether it is required...I don't know? We had a special `ReadOperations.GetAlertSummary` that I considered reusing and it might be fine to consolidate them?

Ah, I see the pattern now (some places call .get and then do their own ensureAuthorized), so let's leave it as is then 👍